pkg/agent/runner.go

```diff
@@ -1661,7 +1666,13 @@ func (r *Runner) doInformantDownscale(ctx context.Context, logger *zap.Logger, t
 		return nil, fmt.Errorf("%s: InformantServer is not set (this should not occur after startup)", msg)
 	}
 
-	resp, err := server.Downscale(ctx, logger, to)
+	// TODO: is it racy to read informantIsMonitor
```

Comment:
maybe

The read must happen while either holding r.lock or r.server.requestLock. `(*InformantServer).Downscale` acquires its request lock, so holding that while making requests here wouldn't work, unfortunately.